pkg/virt-controller/watch/migration.go

```diff
@@ -293,15 +305,31 @@ func copy(migration *kubev1.Migration) (*kubev1.Migration, error) {
 }
 
 // Returns the number of  running pods and if a pod for exactly that migration is currently running
-func investigateTargetPodSituation(migration *kubev1.Migration, podList *k8sv1.PodList) (int, *k8sv1.Pod) {
+func investigateTargetPodSituation(migration *kubev1.Migration, podList *k8sv1.PodList, migrationStore cache.Store) (int, *k8sv1.Pod, error) {
 	var targetPod *k8sv1.Pod = nil
-	for _, pod := range podList.Items {
+	podCount := 0
+	for idx, pod := range podList.Items {
 		if pod.Labels[kubev1.MigrationUIDLabel] == string(migration.GetObjectMeta().GetUID()) {
-			targetPod = &pod
-			break
+			targetPod = &podList.Items[idx]
+			podCount += 1
+			continue
+		}
+		key := fmt.Sprintf("%s/%s", pod.ObjectMeta.Namespace, pod.Labels[kubev1.MigrationLabel])
+		cachedObj, exists, err := migrationStore.GetByKey(key)
```

Comment:
There's an issue here. In a real-world situation we can never obtain the record we would have been looking for. This cache stores migration objects by name. The only object we can get back from the cache *is* the migration object we already passed into this function (because we already looked that object up to get to this function).